### PR TITLE
Users/srnara/mongo index

### PR DESCRIPTION
**Pull request by srinathnarayanan**
Added the option to create Wild card index on all fields while creating a mongo collection

mongo index wildcard

### CODE REVIEW

File: src/Explorer/Panes/AddCollectionPane.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ export default class AddCollectionPane extends ContextualPaneBase {
   public ruToolTipText: ko.Computed<string>;
   public canConfigureThroughput: ko.PureComputed<boolean>;
   public showUpsellMessage: ko.PureComputed<boolean>;
+  public shouldCreateMongoWildcardIndexOnAllFields: ko.Observable<boolean>;
```
Comment (swviswan) [inline src/Explorer/Panes/AddCollectionPane.ts:102]: >shouldCreateMongoWildcardIndexOnAllFields [](start = 9, length = 41)

nit: smaller name? createMongoWildCardIndex?

File: src/Explorer/Panes/AddCollectionPane.html [hunk 2]
```diff
@@ -418,7 +418,6 @@
               </div>
               <!-- large parition key - end -->
 
-              <!-- Provision collection throughput - start -->
```
Comment (swviswan) [inline src/Explorer/Panes/AddCollectionPane.html:421]: > <!-- Provision collecti [](start = 13, length = 24)

?

**Review by swviswan — APPROVED** [review]
:shipit:

File: src/Common/dataAccess/createCollection.ts [hunk 3]
```diff
@@ -182,6 +188,11 @@ const createMongoCollection = async (params: DataModels.CreateCollectionParams):
     params.collectionId,
     rpPayload
   );
+
+  TelemetryProcessor.trace(Action.CreateMongoCollectionWithWildcardIndex, ActionModifiers.Mark, {
```
Comment (swviswan) [inline src/Common/dataAccess/createCollection.ts:192]: >trace [](start = 21, length = 5)

do we have telemetry for all mongo create operations here. so we can easily compute the ratio of people opting in with just the data we have?

Comment (srinathnarayanan) [reply src/Common/dataAccess/createCollection.ts:192]: this was supposed to be only if we params had "createMongoWildcardIndex". will update it. this telemetry is only to trace mongo collection creation where user has opted in to have wild card index on all fields

File: src/Explorer/Explorer.ts [hunk 4]
```diff
@@ -569,6 +555,24 @@ export default class Explorer {
...
 
+    this.isEnableMongoCapabilityPresent = ko.computed(() => {
+      const capabilities = this.databaseAccount && this.databaseAccount()?.properties?.capabilities;
+      if (!capabilities) {
+        return false;
+      }
+
+      for (let i = 0; i < capabilities.length; i++) {
+        if (typeof capabilities[i] === "object") {
+          if (capabilities[i].name === Constants.CapabilityNames.EnableMongo) {
```
Comment (languy) [inline src/Explorer/Explorer.ts:566]: nit: merge the two `if` statements into one.

**Review by languy — APPROVED** [review]

**Pull request merged**